### PR TITLE
Add Bitwarden importer

### LINKED ISSUE #858: Support importing totp from Bitwarden
Bitwarden allows to export json, and csv. It would be good to support importing/stripping the totp codes from these files.
json would be far easier than CSV.

json example: "totp": "JVN3SXK7PFP44WDWSHAP4VVVFUSSGI6N"

### PR DESCRIPTION
**Pull request by CristianAUnisa**
Resolve #858

### CODE REVIEW

**Comment by CristianAUnisa** [comment]
@alexbakker import from a JSON file is already working, I still need to create tests for it; as reported in the linked issue, encrypted json files from Bitwarden should be unimportable (for now).
I'm going to start working on the csv format, but I'd like to ask you a couple of questions:

- Do you think that we should have two separate options on the importer selector (one for the JSON format, the other one for the CSV) or should the app recognise the right format automatically?
- Do you think it would be worth to add a CSV parser library to Aegis? I can work around it in this case but I was wondering if a parser would be useful in the future

**Comment by CristianAUnisa** [comment]
Steam account URIs are prefixed by `steam:// `
https://bitwarden.com/help/authenticator-keys/#steam-guard-totps

I made a custom test for this case

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -377,6 +377,7 @@
     <string name="importer_help_authenticator_plus">Supply an Authenticator Plus export file obtained through <b>Settings -> Backup &amp; Restore -> Export as Text and HTML</b>.</string>
     <string name="importer_help_authy">Supply a copy of <b>/data/data/com.authy.authy/shared_prefs/com.authy.storage.tokens.authenticator.xml</b>, located in the internal storage directory of Authy.</string>
     <string name="importer_help_andotp">Supply an andOTP export/backup file.</string>
+    <string name="importer_help_bitwarden">Supply a Bitwarden export/backup file.</string>
```
Comment (alexbakker) [inline app/src/main/res/values/strings.xml:380]: Could you add a note at the end that says: "Encrypted files are not supported".?

**Review by alexbakker — CHANGES_REQUESTED** [review]
Nice work! Just a couple of comments.

**Comment by CristianAUnisa** [comment]
Just committed your suggestions, let me know if you spot anything wrong! I run the tests again and everything is working.

**Comment by alexbakker** [comment]
Looks good, thanks! Could you squash your changes into a single commit?

**Comment by CristianAUnisa** [comment]
Done!

**Comment by CristianAUnisa** [comment]
I remembered there's a section in the readme file listing supported apps for the importing feature; Bitwarden can be listed when there's a new release

**Review by alexbakker — APPROVED** [review]
Thanks!

**Comment by devnoname120** [comment]
@CristianAUnisa I tried to import from CSV and the importer fails on any TOTP that doesn't explicitly specify the protocol, even though it's optional in Bitwarden.

```
com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null

com.beemdevelopment.aegis.importers.DatabaseImporterEntryException: com.beemdevelopment.aegis.otp.GoogleAuthInfoException: Unsupported protocol: null
```

**Comment by CristianAUnisa** [comment]
Is Bitwarden using the TOTP protocol as the default one when none is specified?

**Comment by devnoname120** [comment]
@CristianAUnisa I can confirm:

> By default, Bitwarden will generate six-digit TOTPs using SHA-1 and rotate them every 30 seconds

([source](https://bitwarden.com/help/authenticator-keys/#:~:text=By%20default%2C%20Bitwarden%20will%20generate,URI%20for%20your%20vault%20item.))

**Pull request merged**